subworkflows/nf-core/fastqc_umitools_trimgalore.nf

```diff
@@ -30,6 +31,26 @@ workflow FASTQC_UMITOOLS_TRIMGALORE {
         UMITOOLS_EXTRACT ( reads ).reads.set { umi_reads }
         umi_log     = UMITOOLS_EXTRACT.out.log
         ch_versions = ch_versions.mix(UMITOOLS_EXTRACT.out.versions.first())
+
+        // Discard R1 / R2 if required
+        if ([1,2].contains(umi_discard_read)) {
+            def keep_idx = 1
+            if (umi_discard_read == 2) {
+                keep_idx = 0
+            }
+
+            UMITOOLS_EXTRACT
+                .out
+                .reads
+                .map { meta, reads ->
+                    if (!meta.single_end) {
+                        meta['single_end'] = true
+                        reads = reads[keep_idx]
```

Comment:
```suggestion
                        reads = reads[umi_discard_read % 2]
```
Maybe this is less readable though, but then you don't need:
```
def keep_idx = 1
if (umi_discard_read == 2) {
    keep_idx = 0
}
```

Reply:
Nice! Added in https://github.com/nf-core/rnaseq/pull/771/commits/e09cde45a7151977d3e7c4880c55e070eb7b727d